### DISCUSSION TITLE
Can i specified rendering mode to deal with vue reactivity core api outside script setup?

### DISCUSSION THREAD

**Discussion by Nemure231** [nuxt/nuxt#19060] [Questions] [answered]
In the Nuxt3 docs, i find vue reactivity api like ref(), reactive() and etc, cannot be place outside script setup. I need to use useState composable for SSR friendly reactive shared state, accourding to: https://nuxt.com/docs/getting-started/state-management

But i find that i can choose different rendering mode with routeRules, accourding to: https://nuxt.com/docs/guide/concepts/rendering#route-rules

My question is, if i change rendering mode to SPA only,  can i use vue reactivity outside script setup?

for example i have composable like this:

```js
export const useMyState = () => {

    const stateData = ref(false)

    const state = computed({
        get: () => stateData.value,
        set: (val) => {
            stateData.value = val
        }
    })

    return {
        state
    }
}

```

then, i write route rules like this:

```js
export default defineNuxtConfig({
  routeRules: {
    // i will be only using composable above just for this page
    '/blog': { ssr: false },
  }
})

```
Is this possible?

**Comment by Baroshem** [comment, marked as answer]
I dont think so.

Ref and other vue utils was meant to be used in defineComponent setup() or script setup.

I would recommend to not do it :)

**Reply by Nemure231** [reply]
Yeah i think i agreed with that, beside i found another safe solution. Thanks for your answer.
